beacon-chain/rpc/beaconv1/state.go

```diff
@@ -43,7 +46,49 @@ func (bs *Server) GetGenesis(ctx context.Context, _ *ptypes.Empty) (*ethpb.Genes
 
 // GetStateRoot calculates HashTreeRoot for state with given 'stateId'. If stateId is root, same value will be returned.
 func (bs *Server) GetStateRoot(ctx context.Context, req *ethpb.StateRequest) (*ethpb.StateRootResponse, error) {
-	return nil, errors.New("unimplemented")
+	ctx, span := trace.StartSpan(ctx, "beaconv1.GetStateRoot")
+	defer span.End()
+
+	var (
+		stateIdString = strings.ToLower(string(req.StateId))
+		root          []byte
+		err           error
+	)
+
+	switch stateIdString {
+	case "head":
+		root, err = bs.headStateRoot(ctx)
+	case "genesis":
+		root, err = bs.genesisStateRoot(ctx)
+	case "finalized":
+		root, err = bs.finalizedStateRoot(ctx)
+	case "justified":
+		root, err = bs.justifiedStateRoot(ctx)
+	default:
+		ok, matchErr := regexp.Match("0x[0-9a-fA-F]{64}", []byte(hexutil.Encode(req.StateId)))
```

Comment:
Let's abstract this perhaps into a hex or bytesutil shared package, it's a magic regex that makes sense but would be nicer as a small helper function that can be unit tested